iocage/lib/Datasets.py

```diff
@@ -0,0 +1,146 @@
+import helpers
+
+import libzfs
+
+
+class Datasets:
+
+    ZFS_POOL_ACTIVE_PROPERTY = "org.freebsd.ioc:active"
+
+    def __init__(self, root=None, pool=None, zfs=None, logger=None):
+        helpers.init_logger(self, logger)
+        helpers.init_zfs(self, zfs)
+
+        self._datasets = {}
+
+        if isinstance(root, libzfs.ZFSDataset):
+            self.root = root
+            return
+
+        if isinstance(pool, libzfs.ZFSPool):
+            self.root = self._get_or_create_dataset(
+                "iocage",
+                root_name=pool.name,
+                pool=pool
+            )
+            return
+
+        active_pool = self.active_pool
+
+        if active_pool is None:
+            msg = ("iocage is not activated yet - "
+                   "please run `iocage activate` first and select a pool")
+            self.logger.error(msg)
+            raise Exception(msg)
+        else:
+            self.root = self.zfs.get_dataset(f"{active_pool.name}/iocage")
+                
+
+    @property
+    def active_pool(self):
+        prop = self.ZFS_POOL_ACTIVE_PROPERTY
+        for pool in self.zfs.pools:
+            try:
+                if pool.root_dataset.properties[prop].value == "yes":
+                    return pool
+            except:
+                pass
+        return None
+
+    @property
+    def releases(self):
+        return self._get_or_create_dataset("releases")
+
+    @property
+    def base(self):
+        return self._get_or_create_dataset("base")
+
+    @property
+    def jails(self):
+        return self._get_or_create_dataset("jails")
+
+    def activate(self):
+        self.activate_pool(self.root.pool)
+
+    def activate_pool(self, zfs_pool):
+
+        prop = self.ZFS_POOL_ACTIVE_PROPERTY
+
+        is_pool_already_active = False
+        try:
+            if zfs_pool.root_dataset.properties[prop].value == "yes":
+                is_pool_already_active = True
+        except:
+            pass
```

Comment:
same here. We want to ignore the exception in case the property `org.freebsd.ioc:active` is not set on the pool's root dataset.